src/Features/Core/Portable/Diagnostics/DiagnosticAnalyzerService_IncrementalAnalyzer.cs

```diff
@@ -128,6 +129,7 @@ public override void RemoveProject(ProjectId projectId)
             public override bool NeedsReanalysisOnOptionChanged(object sender, OptionChangedEventArgs e)
             {
                 return e.Option.Feature == nameof(SimplificationOptions) ||
+                       e.Option.Feature == nameof(CodeStyleOptions) ||
```

Comment:
Or, at the very least, this should operate in the reverse manner.  If we want to not reanalyze in some cases, then we should enumerate those cases .  Right now this introduces tight coupling between disparate components of the system in an undiscoverable and undocumented manner.